### PR TITLE
Offspring seed saves

### PR DESCRIPTION
**Pull request by AceXP**
- What is the purpose of the pull request?

Combine PR #1559 OffspringSeed with PR #1561 Save folders

- Give a brief description of what you changed or added.

Offspring is stored in a seperate OffspringSeed class, seperate from the NPC class. Only when offspring is encountered (in an area or with the offspring map) is an NPC created from the offspringSeed. This reduces the save file size as less data is stored. And it increases performance as the number of NPC's to loop over is reduced.

When loading an old save, any existing NPCOffspring NPC's that have not been encountered yet are converted to the new offspringSeed class. This is a one-time only event. It would be best to make a new save straight away. But also see the advise below.

Also some minor tweaks for dialogue when offspring is not birthed by the player, but by an incubator.

Converts save files into save folders, NPC's and offspringSeed are each saved in their own file. Since offspringSeed only needs to be saved once (they will never change) this reduces save time.

- Are any new graphical assets required?

No

- Has this change been tested? If so, mention the version number that the test was based on.

Yes, tested with 0.4.0.10 (and earlier versions). Although the recommendation is to do more testing and STRONGLY recommend users to back up their saves when this gets added. There are multiple changes in the way saves work with these changes.

- So we have a better idea of who you are, what is your Discord Handle?

AceXP #0930 and Maxis010 #3060

### CODE REVIEW

File: src/com/lilithsthrone/game/Game.java [hunk 1]
```diff
@@ -518,22 +513,17 @@ public static void exportGame(String exportFileName, boolean allowOverwrite) {
...
+		if(dir.exists()) {
+			if(!allowOverwrite) {
+				Main.game.flashMessage(PresetColour.GENERIC_BAD, "Save already exists!");
+				return;
+			} else {
+				overwrite = true;
 			}
 		}
-
+		dir.mkdir();
```
Comment (Maxis010) [inline src/com/lilithsthrone/game/Game.java]: Looks like this slipped through when I was experimenting with per character folders

File: src/com/lilithsthrone/game/Game.java [hunk 2]
```diff
@@ -655,46 +645,75 @@ public static void exportGame(String exportFileName, boolean allowOverwrite) {
...
 		}
-
-
+		
+		// Add all offspringSeed:
+		try {
+			String id = "";
+			for(OffspringSeed offspringSeed : Main.game.getOffspringSeedMap().values()) {
+				id = offspringSeed.getId().replace(",", "_");
+				if(!new File("data/saves/"+exportSaveName+"/"+id+".xml").exists()) {
```
Comment (Maxis010) [inline src/com/lilithsthrone/game/Game.java:700]: I assume offspringSeed won't change after creation so it's safe to only have an exists() check?

File: src/com/lilithsthrone/game/Game.java (hunk 1 above)
Comment (AceXP) [reply src/com/lilithsthrone/game/Game.java]: This probably stems from the merge and needs to be looked into.

File: src/com/lilithsthrone/game/Game.java (hunk 2 above)
Comment (AceXP) [reply src/com/lilithsthrone/game/Game.java:700]: Yes

**Pull request merged**